spec/user_repository_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'user_repository'
+require 'user'
+require 'bcrypt'
+
+def reset_users_table
+    seed_sql = File.read('spec/seeds/test_seeds.sql')
+    connection = PG.connect({ host: '127.0.0.1', dbname: 'chitter_test' })
+    connection.exec(seed_sql)
+  end
+  
+  describe UserRepository do
+    before(:each) do 
+      reset_users_table
+    end
+
+  context 'all' do
+    it 'retrieves all users and returns a list of them' do
+      repo = UserRepository.new
+      users = repo.all
+
+      expect(users.length).to eq(2)
+      expect(users.first.name).to eq('Alex')
+
+      expect(users.last.name).to eq('Zeus')
+    end
+  end
+
+  context 'create' do
+    it 'creates a new user' do
+      repo = UserRepository.new
+      new_user = User.new
+
+      new_user.id = '3'
+      new_user.name = 'Max'
+      new_user.email = 'maxemail@test.com'
+      new_user.username = 'maxonthesax'
+      new_user.password = 'password123'
+
+      repo.create(new_user)
+
+      all_users = repo.all
+      newest_user = all_users.last
+      
+      expect(all_users.length).to eq(3)
+      expect(newest_user.id).to eq '3'
+      expect(newest_user.name).to eq 'Max'
+      expect(newest_user.email).to eq 'maxemail@test.com'
+      expect(newest_user.username).to eq 'maxonthesax'
+      # not sure how to test for password (as output will be different each time)
```

Comment:
This is a great opportunity for mocking, not that I know how to do it though aha